lib/urldata.h

```diff
@@ -892,6 +892,21 @@ struct proxy_info {
   char *passwd;  /* proxy password string, allocated */
 };
 
+/* struct for HTTP CONNECT state data */
+struct http_connect_state {
+  char *connect_buffer;
+  int perline; /* count bytes per line */
+  int keepon;
+  char *line_start;
+  char *ptr; /* where to store more data */
+  curl_off_t cl; /* size of content to read and ignore */
+  enum {
+    TUNNEL_INIT,    /* init/default/no tunnel state */
+    TUNNEL_CONNECT, /* CONNECT has been sent off */
+    TUNNEL_COMPLETE /* CONNECT response received completely */
+  } tunnel_state;   /* two separate ones to allow FTP */
```

Comment:
It is indeed! Thanks. The updated logic allocates a fresh struct for each CONNECT...